client/v2/client.go

```diff
@@ -0,0 +1,371 @@
+		useragent: conf.UserAgent,
+		httpClient: &http.Client{
+			Timeout:   conf.Timeout,
+			Transport: tr,
+		},
+	}
+}
+
+type client struct {
+	url        *url.URL
+	username   string
+	password   string
+	useragent  string
+	httpClient *http.Client
+}
+
+// BatchPoints is an interface into a batched grouping of points to write into
+// InfluxDB together.
+type BatchPoints interface {
+	// AddPoint adds the given point to the Batch of points
+	AddPoint(p *Point)
+	// Points lists the points in the Batch
+	Points() []*Point
+
+	// Precision returns the currently set precision of this Batch
+	Precision() string
+	// SetPrecision sets the precision of this batch.
+	SetPrecision(s string)
+
+	// Database returns the currently set database of this Batch
+	Database() string
+	// SetDatabase sets the database of this Batch
+	SetDatabase(s string)
+
+	// WriteConsistency returns the currently set write consistency of this Batch
+	WriteConsistency() string
+	// SetWriteConsistency sets the write consistency of this Batch
+	SetWriteConsistency(s string)
+
+	// RetentionPolicy returns the currently set retention policy of this Batch
+	RetentionPolicy() string
+	// SetRetentionPolicy sets the retention policy of this Batch
+	SetRetentionPolicy(s string)
+}
+
+// NewBatchPoints returns a BatchPoints interface based on the given config.
+func NewBatchPoints(c BatchPointsConfig) BatchPoints {
+	return &batchpoints{
+		database:         c.Database,
+		precision:        c.Precision,
+		retentionPolicy:  c.RetentionPolicy,
+		writeConsistency: c.WriteConsistency,
+	}
+}
+
+type batchpoints struct {
+	sync.Mutex
+
+	points           []*Point
+	database         string
+	precision        string
+	retentionPolicy  string
+	writeConsistency string
+}
+
+func (bp *batchpoints) AddPoint(p *Point) {
+	bp.Lock()
+	defer bp.Unlock()
+	bp.points = append(bp.points, p)
+}
+
+func (bp *batchpoints) Points() []*Point {
+	bp.Lock()
+	defer bp.Unlock()
+	return bp.points
+}
+
+func (bp *batchpoints) Precision() string {
+	bp.Lock()
+	defer bp.Unlock()
+	return bp.precision
+}
+
+func (bp *batchpoints) Database() string {
+	bp.Lock()
+	defer bp.Unlock()
+	return bp.database
+}
+
+func (bp *batchpoints) WriteConsistency() string {
+	bp.Lock()
+	defer bp.Unlock()
+	return bp.writeConsistency
+}
+
+func (bp *batchpoints) RetentionPolicy() string {
+	bp.Lock()
+	defer bp.Unlock()
+	return bp.retentionPolicy
+}
+
+func (bp *batchpoints) SetPrecision(p string) {
+	bp.Lock()
+	defer bp.Unlock()
+	bp.precision = p
+}
+
+func (bp *batchpoints) SetDatabase(db string) {
+	bp.Lock()
+	defer bp.Unlock()
+	bp.database = db
+}
+
+func (bp *batchpoints) SetWriteConsistency(wc string) {
+	bp.Lock()
+	defer bp.Unlock()
+	bp.writeConsistency = wc
+}
+
+func (bp *batchpoints) SetRetentionPolicy(rp string) {
+	bp.Lock()
+	defer bp.Unlock()
+	bp.retentionPolicy = rp
+}
+
+type Point struct {
+	pt models.Point
+}
+
+// NewPoint returns a point with the given timestamp
+func NewPoint(
```

Comment:
Not sure how we feel about this, but if the function signature was the following:

`NewPoint(name string, tags map[string]string, fields map[string]interface{}, time.Time ....)`

supplying the time would then be optional. You wouldn't need `NewPointWithoutTime`. This may be an easier-to-use API.